File: project2/Team3/server/src/main/java/bitc/full502/lostandfound/service/BoardService.java
package bitc.full502.lostandfound.service;

import bitc.full502.lostandfound.dto.BoardDTO;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDateTime;
import java.util.List;

public interface BoardService {
    //    목록 불러오기
    List<BoardDTO> getAllBoardList() throws Exception;

    //    게시글 수정(이미지 포함)
    BoardDTO updateBoardWithImage(BoardDTO dto, MultipartFile file) throws Exception;

    //    게시글 삭제
    void deleteBoard(Long id) throws Exception;

    //    게시글 검색
    List<BoardDTO> searchBoardList(String keyword, Integer categoryId, String type, LocalDateTime fromDate, LocalDateTime toDate) throws Exception;

    //    게시글 등록
    BoardDTO insertBoard(BoardDTO dto) throws Exception;

    BoardDTO getBoardDetail(Long id) throws Exception;

    void updateBoardStatus(Long idx) throws Exception;
}

File: project2/Team3/client/app/src/main/java/bitc/full502/lostandfound/data/api/BoardService.kt
package bitc.full502.lostandfound.data.api

import bitc.full502.lostandfound.data.model.BoardData
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Part
import retrofit2.http.Path
import retrofit2.http.Query

interface BoardService {
    @GET("board/list")
    fun getAllBoardList(): Call<List<BoardData>>

    //    controller 부분이 멀티파트라서 꼭 파트로 보내야함
    @Multipart
    @POST("board/insert")
    fun insertBoard(
        @Header("Authorization") token: String,
        @Part("dto") dtoJson: RequestBody,
        @Part file: MultipartBody.Part?
    ): Call<BoardData>

    @GET("board/{id}")
    fun getBoardDetail(
        @Path("id") id: Long
    ): Call<BoardData>


    @Multipart
    @PUT("board/update")
    fun updateBoard(
        @Part("dto") dto: RequestBody,
        @Part file: MultipartBody.Part?
    ): Call<BoardData>

    @PUT("board/{id}/complete")
    fun completeBoard(
        @Path("id") id: Long,
    ):Call<Unit>

    @GET("board/search")
    fun searchBoard (
        @Query("keyword") keyword: String?,
        @Query("categoryId") categoryId: Int?,
        @Query("type") type: String?,
        @Query("fromDate") fromDate: String?,
        @Query("toDate") toDate: String?
    ): Call<List<BoardData>>

}
